src/scroll-seek-controls.js

```diff
@@ -0,0 +1,178 @@
+// src/scroll-seek-controls.js
+
+import { configAddChangeListener, configRead } from './config.js';
+import { requireElement } from './player-api.ts';
+
+/**
+ * ScrollSeek class to handle scroll seeking functionality in video players.
+ * This class listens for wheel events to change the video's seek position
+ * based on the scroll direction and updates the UI accordingly.
+ */
+
+class ScrollSeek {
+  #video = null;
+  #container = null;
+  #initialized = false;
+  #observer = null;
+  #elPlayheadStyle = null;
+  #elProgressBarStyle = null;
+  #scrollAttached = false;
+  #wasFocused = null;
+  #containerWidth = 1000;
+  #opts = { passive: false, capture: true };
+  #videoEvents = ['timeupdate', 'ended'];
+
+  constructor() {
+    if (configRead('enableScrollSeek')) this.enable();
+    configAddChangeListener('enableScrollSeek', (e) =>
+      e.detail?.newValue ? this.enable() : this.disable()
+    );
+  }
+
+  #getStep = () =>
+    Math.min(100, Math.max(3, (this.#video?.duration || 600) * 0.03)); // 3% of video duration, capped between 3–100 seconds
+
+  #updateUI = () => {
+    if (!this.#video?.duration || !isFinite(this.#video.currentTime)) return;
+
+    const pct = this.#video.currentTime / this.#video.duration;
+    const px = this.#containerWidth * pct;
+
+    this.#elPlayheadStyle.transform = `translateX(${px}px)`;
+    this.#elProgressBarStyle.transform = `translateX(0) scaleX(${pct})`;
+  };
+
+  #handleScroll = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    e.stopImmediatePropagation();
+
+    if (!this.#video?.duration) return;
+
+    const step = this.#getStep();
+    const direction = e.deltaY > 0 ? -step : step;
+
+    this.#video.currentTime = Math.min(
+      Math.max(this.#video.currentTime + direction, 0),
+      this.#video.duration
+    );
+
+    this.#updateUI();
+  };
+
+  #attachScroll = () => {
+    if (!this.#scrollAttached) {
+      window.addEventListener('wheel', this.#handleScroll, this.#opts);
+      this.#scrollAttached = true;
+    }
+  };
+
+  #detachScroll = () => {
+    if (this.#scrollAttached) {
+      window.removeEventListener('wheel', this.#handleScroll, this.#opts);
+      this.#scrollAttached = false;
+    }
+  };
+
+  #checkFocus = () => {
+    const isFocused = this.#container?.classList.contains(
+      'ytLrProgressBarFocused'
+    );
+    if (isFocused !== this.#wasFocused) {
+      this.#wasFocused = isFocused;
+      isFocused ? this.#attachScroll() : this.#detachScroll();
+    }
+  };
+
+  #watchProgressBarFocus = () => {
+    if (!this.#container) return;
+
+    this.#checkFocus();
+
+    this.#observer?.disconnect();
+    this.#observer = new MutationObserver(this.#checkFocus);
+    this.#observer.observe(this.#container, {
```

Comment:
I still don't understand why you need to disconnect and reconnect the observer. You already disconnect it in `disable()`, and `this.#container` can only possibly change with `enable()`.